pmpro-approvals.php

```diff
@@ -1044,12 +1094,17 @@ public static function pmpro_before_change_membership_level( $level_id, $user_id
 	/**
 	 * Send an email to an admin when a user has signed up for a membership level that requires approval.
 	 */
-	public static function pmpro_after_change_membership_level( $level_id, $user_id ) {
+	public static function pmpro_after_change_membership_level( $level_id, $user_id, $cancel_level ) {
 
 		//check if level requires approval, if not stop executing this function and don't send email.
-		if ( ! self::requiresApproval( $level_id ) ) {
+		if ( ! self::requiresApproval( $level_id ) && ! self::requiresApproval( $cancel_level ) ) {
+			return;
+		}
+
+		if ( $cancel_level ) {
+			self::removeMember( $user_id, $cancel_level );
 			return;
-		}		
+		}
```

Comment:
> we may need user to stay as approved for signing up for a different level.

@andrewlimaza Could this suffice for the scenario you mentioned above? This worked well for my test scenarios but I feel I may have missed one or two other possibilities.